src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -85,6 +88,19 @@ public static class SqlServerBaseRegistrationExtensions
                 .AsSelf()
                 .AsImplementedInterfaces();
 
+            switch (config.AuthenticationType)
+            {
+                case SqlServerAuthenticationType.ManagedIdentity:
+                    services.AddSingleton<ISqlConnectionFactory, ManagedIdentitySqlConnectionFactory>();
+                    services.AddSingleton<IAccessTokenHandler, ManagedIdentityAccessTokenHandler>();
```

Comment:
If this is supposed to be reusabled, maybe this should always be registered

Reply:
Not necessarily. For now, we just fetch token if authentication type is managed identity. However, in future, if this is to be used for something else as well, we can register it without this check.